test/extended/nto_util.go

```diff
@@ -0,0 +1,488 @@
+		Logf("machinesetName is %v in choseOneWorkerNodeToRunCase", machinesetName)
+
+		if len(machinesetName) != 0 {
+			machinesetReplicas, err := oc.AsAdmin().WithoutNamespace().Run("get").Args("machineset", "-n", "openshift-machine-api", machinesetName, "-o=jsonpath={.spec.replicas}").Output()
+			o.Expect(err).NotTo(o.HaveOccurred())
+			if !strings.Contains(machinesetReplicas, "0") {
+				// First check if any nodes exist with this machineset label
+				nodeNames, err := oc.AsAdmin().WithoutNamespace().Run("get").Args("nodes", "-l", "machine.openshift.io/cluster-api-machineset="+machinesetName, "-o=jsonpath={.items[*].metadata.name}").Output()
+				o.Expect(err).NotTo(o.HaveOccurred())
+				if len(strings.TrimSpace(nodeNames)) > 0 {
+					// Get the first node name
+					nodeList := strings.Fields(nodeNames)
+					tunedNodeName = nodeList[0]
+					o.Expect(tunedNodeName).NotTo(o.BeEmpty())
+				} else {
+					Logf("No nodes found for machineset %s, falling back to node selection without machineset", machinesetName)
+					tunedNodeName = choseOneWorkerNodeNotByMachineset(oc, choseBy)
+				}
+			} else {
+				tunedNodeName = choseOneWorkerNodeNotByMachineset(oc, choseBy)
+			}
+		} else {
+			tunedNodeName = choseOneWorkerNodeNotByMachineset(oc, choseBy)
+		}
+	} else {
+		tunedNodeName = choseOneWorkerNodeNotByMachineset(oc, choseBy)
+		Logf("the tunedNodeName that we get inside choseOneWorkerNodeToRunCase when choseBy %v is %v ", choseBy, tunedNodeName)
+	}
+	return tunedNodeName
+}
+
+// getLinuxWorkerMachinesets returns a list of Linux worker machinesets, filtering out Windows and edge nodes
+func getLinuxWorkerMachinesets(oc *CLI) []string {
+	machinesetList, err := oc.AsAdmin().WithoutNamespace().Run("get").Args("-n", "openshift-machine-api", "machineset", "-ojsonpath={.items[*].metadata.name}").Output()
+	o.Expect(err).NotTo(o.HaveOccurred())
+
+	workerMachineSets := strings.Split(machinesetList, " ")
+	Logf("workerMachineSets is %v", workerMachineSets)
+
+	linuxMachineset := make([]string, 0, len(workerMachineSets))
+	for _, machineset := range workerMachineSets {
+		// Skip windows and edge nodes
+		if strings.Contains(machineset, "windows") || strings.Contains(machineset, "edge") {
+			Logf("skip windows or edge node [ %v ]", machineset)
+		} else if len(machineset) > 0 {
+			linuxMachineset = append(linuxMachineset, machineset)
+		}
+	}
+	Logf("linuxMachineset is %v", linuxMachineset)
+	return linuxMachineset
+}
+
+func getWorkerMachinesetName(oc *CLI, machineseetSN int) string {
+	linuxMachineset := getLinuxWorkerMachinesets(oc)
+
+	var machinesetName string
+	if machineseetSN < len(linuxMachineset) {
+		machinesetName = linuxMachineset[machineseetSN]
+	}
+
+	Logf("machinesetName is %v in getWorkerMachinesetName", machinesetName)
+	return machinesetName
+}
+
+func choseOneWorkerNodeNotByMachineset(oc *CLI, choseBy int) string {
+	//0 means the first worker node, 1 means the last worker node
```

Comment:
This comment looks out-of-place.  Either move to the `switch choseBy {` or add please more text to indicate what 0/1 refers to (`choseBy`).  Also, can we have some consistency when commenting?  Why have some comments space after `//` and some do not?